examples/galaxies/plot_ellipticity.py

```diff
@@ -0,0 +1,123 @@
+"""
+Galaxy Ellipticity Distributions
+================================
+
+This example demonstrate how to sample ellipticity distributions
+in SkyPy.
+
+"""
+
+
+# %%
+# 3D Ellipticity Distribution
+# ---------------------------
+#
+# In Ryden 2004 [1]_, the ellipticity is sampled by randomly projecting
+# a 3D ellipsoid with principal axes :math:`A > B > C`.
+#
+# The distribution of the axis ratio :math:`\gamma = C/A` is a truncated
+# normal with mean :math:`\mu_\gamma` and standard deviation
+# :math:`\sigma_\gamma`.
+#
+# The distribution of :math:`\epsilon = \log(1 - B/A)` is truncated normal
+# with mean :math:`\mu` and standard deviation :math:`\sigma`.
+#
+#
+# In ``SkyPy``, we can sample 3D ellipticities using the
+# :func:`skypy.galaxies.morphology.ryden04_ellipticity()` distribution
+# described above [1].
+#
+# Validation plot with SDSS Data
+# ------------------------------
+#
+# Here we validate our function by comparing our ``SkyPy`` simulated galaxy
+# ellipticities against observational data from SDSS DR1 in Figure 1 [1].
+# You can download the data file
+# :download:`SDSS_ellipticity <../../../examples/galaxies/SDSS_ellipticity.txt>`,
+# stored as a 2D array: ellipticity_X, ellipticity_T.
+#
+
+import numpy as np
+
+# Load SDSS data from Fig. 1 in Ryden 2004
+eX, eT = np.loadtxt('SDSS_ellipticity.txt', unpack=True)
+Ngal = len(eX)
+e = np.hypot(eX, eT)
+q_amSDSS = np.sqrt((1 - e)/(1 + e))
+
+
+# %%
+#
+# In this example, we generate 100 galaxy ellipticity samples using the
+# ``SkyPy`` function
+# :func:`skypy.galaxies.morphology.ryden04_ellipticity()` and the
+# best fit parameters
+# given in Ryden 2004 [1]:
+# :math:`\mu_\gamma =0.222`, :math:`\sigma_\gamma=0.057`, :math:`\mu =-1.85`
+# and :math:`\sigma=0.89`.
+#
+
+from skypy.galaxies.morphology import ryden04_ellipticity
+
+# Best fit parameters from Fig. 1 in Ryden 2004
+mu_gamma, sigma_gamma, mu, sigma = 0.222, 0.057, -1.85, 0.89
+
+# Binning scheme of Fig. 1
+bins = np.linspace(0, 1, 41)
+mid = 0.5 * (bins[:-1] + bins[1:])
+
+# Mean and variance of sampling
+mean = np.zeros(len(bins)-1)
+var = np.zeros(len(bins)-1)
+
+# %%
+#
+
+# Create 100 SkyPy realisations
+for i in range(100):
+    # sample ellipticity
+    e = ryden04_ellipticity(mu_gamma, sigma_gamma, mu, sigma, size=Ngal)
+    # recover axis ratio from ellipticity
+    q = (1 - e)/(1 + e)
+    # bin
+    n, _ = np.histogram(q, bins=bins)
+
+    # update mean and variance
+    x = n - mean
+    mean += x/(i+1)
+    y = n - mean
+    var += x*y
```

Comment:
Is there a nice and short proof of the calculation of the variance in this way or can someone explain it intuitively? I cannot get my head around this and if I try to calculate it, I always get a really long and messy calculation.

Reply:
Don't derive the formula, just show it works.

Let `mu` be the mean of `x` and let `mu2` be the mean of `x*x`, and hence `var = mu2 - mu*mu`.

```
mu[n+1] = (n*mu[n] + x)/(n+1)
        = ((n+1)*mu[n] + x - mu[n])/(n+1)
        = mu[n] + (x - mu[n])/(n+1)

mu2[n+1] = mu2[n] + (x*x - mu2[n])/(n+1)

var[n] + ((x - mu[n])*(x - mu[n+1]) - var[n])/(n+1)
= var[n] + ((x - mu[n])*(x - mu[n+1]) - mu2[n] + mu[n]*mu[n])/(n+1)
= var[n] + (x*x + mu[n]*mu[n+1] - x*mu[n] - x*mu[n+1] - mu2[n] + mu[n]*mu[n])/(n+1)
= var[n] + (x*x - mu2[n])/(n+1) - (mu[n+1] + mu[n])*(x - mu[n])/(n+1)
= var[n] + mu2[n+1] - mu2[n] - (mu[n+1] + mu[n])*(mu[n+1] - mu[n])
= var[n] + mu2[n+1] - mu2[n] - mu[n+1]*mu[n+1] + mu[n]*mu[n]
= var[n+1] ✔
```

Reply:
Brilliant! Thanks for that. But one follow up question:
> 
> 
> var[n] + ((x - mu[n])*(x - mu[n+1]) - var[n])/(n+1)

Where is the second `var[n]` coming from?
According to the code, shouldn't it be
`var[n] + ((x - mu[n])*(x - mu[n+1]))/(n+1)` ?

Reply:
The code could do better because it actually computes `(n+1) var[n+1]` from `n var[n]`, that's why there is a final division at the end. In which case you don't get the `-var` term.

```
(n var[n]) + (x - mu[n])*(x - mu[n+1])
= (n+1) ( var[n] + ((x - mu[n])*(x - mu[n+1]) - var[n])/(n+1) )
= (n+1) var[n+1] ✔
```

Reply:
Ok got it. Thanks :)